simpleGraph/bytesAlign.go

```diff
@@ -0,0 +1,422 @@
+					deletion := &Node{Id: uint32(len(genome.Nodes)), Name: chr.Name, SeqTwoBit: dnaTwoBit.NewTwoBit(dna.StringToBases(vcfsChr[i].Ref)), Prev: nil, Next: nil, Info: &Annotation{Allele: 0, Start: uint32(vcfsChr[i].Pos), Variant: 3}}
+					AddNode(genome, deletion)
+					AddEdge(currMatch, deletion, 1)
+					if strings.Contains(vcfsChr[i].Id, "pbsv") {
+						index = common.MinInt64(vcfsChr[i].Pos+int64(len(deletion.Seq))-1, vcfsChr[i+1].Pos-1)
+					} else {
+						index = vcfsChr[i].Pos + int64(len(deletion.Seq))
+					}
+				} else if isINV(vcfsChr[i]) {
+					currMatch.Seq = append(currMatch.Seq, dna.StringToBases(vcfsChr[i].Ref)...)
+					inversion := mkInversionNode(genome, vcfsChr[i], chr)
+					AddNode(genome, inversion)
+					AddEdge(currMatch, inversion, 1)
+					index = getSvEnd(vcfsChr[i])
+				} else if isCNV(vcfsChr[i]) || isDup(vcfsChr[i]) {
+					currMatch.Seq = append(currMatch.Seq, dna.StringToBases(vcfsChr[i].Ref)...)
+					copyVariance := &Node{Id: uint32(len(genome.Nodes)), Name: chr.Name, SeqTwoBit: dnaTwoBit.NewTwoBit(chr.Seq[vcfsChr[i].Pos:getSvEnd(vcfsChr[i])]), Prev: nil, Next: nil, Info: &Annotation{Allele: 1, Start: uint32(vcfsChr[i].Pos), Variant: 6}}
+					AddNode(genome, copyVariance)
+					AddEdge(currMatch, copyVariance, 1)
+					index = getSvEnd(vcfsChr[i])
+				} else if isHaplotypeBlock(vcfsChr[i]) {
+					refAllele = &Node{Id: uint32(len(genome.Nodes)), Name: chr.Name, SeqTwoBit: dnaTwoBit.NewTwoBit(dna.StringToBases(vcfsChr[i].Ref)), Prev: nil, Next: nil, Info: &Annotation{Allele: 0, Start: uint32(vcfsChr[i].Pos), Variant: 4}}
+					AddNode(genome, refAllele)
+					AddEdge(currMatch, refAllele, 1)
+					altAllele = &Node{Id: uint32(len(genome.Nodes)), Name: chr.Name, SeqTwoBit: dnaTwoBit.NewTwoBit(dna.StringToBases(vcfsChr[i].Alt)), Prev: nil, Next: nil, Info: &Annotation{Allele: 1, Start: uint32(vcfsChr[i].Pos), Variant: 4}}
+					AddNode(genome, altAllele)
+					AddEdge(currMatch, altAllele, 1)
+					index = common.MinInt64(vcfsChr[i].Pos+int64(len(refAllele.Seq))-1, vcfsChr[i+1].Pos-1)
+					currMatch = refAllele
+				}
+				lastMatch = currMatch
+			} else {
+				//num++
+				//log.Printf("index=%d, vcfPos=%d\ndiff=%d\n%s", index, vcfsChr[i].Pos, vcfsChr[i].Pos-index, vcfsChr[i].Info)
+			}
+		}
+	}
+	//Case: last node
+	lastNode := &Node{Id: uint32(len(genome.Nodes)), Name: chr.Name, SeqTwoBit: dnaTwoBit.NewTwoBit(chr.Seq[index:]), Prev: make([]*Edge, 0, len(prev)), Next: nil, Info: &Annotation{Allele: 0, Start: uint32(index + 1), Variant: 0}}
+	AddNode(genome, lastNode)
+	for edge = 0; edge < len(lastMatch.Next); edge++ {
+		AddEdge(lastMatch.Next[edge].Dest, lastNode, 1)
+	}
+	if vcf.Snp(vcfsChr[len(vcfsChr)-2]) || isHaplotypeBlock(vcfsChr[len(vcfsChr)-2]) {
+		AddEdge(altAllele, lastNode, 1)
+	}
+	AddEdge(lastMatch, lastNode, 1)
+	SetEvenWeights(lastMatch)
+	return genome
+}
+
+func checkAlignment(aln *giraf.Giraf, genome *SimpleGraph) bool {
```

Comment:
It might be good to split apart this file into those functions that deal with making the graph, and those that deal with aligning to the graph